### PR TITLE
[MLIR][BUILD] Fix Unicode build issue on Windows.

### PR DESCRIPTION
**Pull request by harrisonGPU**
This issue is from https://github.com/llvm/llvm-project/pull/77668. I encountered a build issue because it used Unicode. When I built MLIR on Windows with Visual Studio 2022, I faced a build failure.

### CODE REVIEW

File: mlir/tools/mlir-rewrite/mlir-rewrite.cpp [hunk 1]
```diff
@@ -320,25 +320,25 @@ LogicalResult markRanges(RewritePad &rewriteState, raw_ostream &os) {
...
 
     if (isGeneric(it)) {
-      rewriteState.insertText(nameRange.Start, "〖");
-      rewriteState.insertText(nameRange.End, "〗");
+      rewriteState.insertText(nameRange.Start, " [");
+      rewriteState.insertText(nameRange.End, "] ");
     } else {
-      rewriteState.insertText(nameRange.Start, "〔");
-      rewriteState.insertText(nameRange.End, "〕");
+      rewriteState.insertText(nameRange.Start, " [");
```
Comment (jpienaar) [inline mlir/tools/mlir-rewrite/mlir-rewrite.cpp:332]: Lets perhaps make this something like " !\[" and "\]! " - previously one could differentiate between generic or not form.

**Review by jpienaar — APPROVED** [review]
SG from me

File: mlir/tools/mlir-rewrite/mlir-rewrite.cpp (hunk 1 above)
Comment (harrisonGPU) [reply mlir/tools/mlir-rewrite/mlir-rewrite.cpp]: Okay, I have already updated it.

**Comment by harrisonGPU** [comment]
Hi @jpienaar , I have updated this PR, please take a look again. :)

**Pull request merged**